Week 3

connienguyen opened the pull request:
Temporary PR for commenting purposes only

connienguyen committed:
Set up XIBs for EventTableViewController

Event Detail page presented as a `UITableViewController` so that it can
easily be customized according to the event (show and hide cells).
+ Set up XIB for `EventImageCell`
+ Set up XIB for `RegisteredCell`
+ Fix lint warnings for `// MARK` comments

connienguyen committed:
Set up XIBs for EventTableViewController

Event Detail page presented as a `UITableViewController` so that it can
easily be customized according to the event (show and hide cells).
+ Set up XIB for `EventImageCell`
+ Set up XIB for `RegisteredCell`
+ Fix lint warnings for `// MARK` comments

connienguyen committed:
Create initial UI and services for viewing event details

+ Event details shown as `EventTableViewController`
+ Set up view controller in storyboard for Event Registration
+ Create `ETouchesAPIService` to manage retrieving and event data from
eTouches API

connienguyen committed:
Create initial UI and services for viewing event details

+ Event details shown as `EventTableViewController`
+ Set up view controller in storyboard for Event Registration
+ Create `ETouchesAPIService` to manage retrieving and event data from
eTouches API

connienguyen committed:
Fix merge conflicts from rebase from login-manual

Files added here are due to merge conflict from rebase and is not new
code.

connienguyen committed:
Create new view controller for displaying event details

+ Added new `Errors` for UI and eTouches related errors
+ Set up XIB and ViewController for `EventDetailViewController
+ `UIViewController` extension for showing and hiding an activity
indicator
+ Added convenience init for `Event`

connienguyen committed:
Create new view controller for displaying event details

+ Added new `Errors` for UI and eTouches related errors
+ Set up XIB and ViewController for `EventDetailViewController
+ `UIViewController` extension for showing and hiding an activity
indicator
+ Added convenience init for `Event`

connienguyen committed:
Configured EventDetailViewController

+ Show EventDetailViewController, configured according to event data
+ Change `AppTransportSecurity` settings to allow arbitrary loads for
event and sponsored images

connienguyen committed:
Configured EventDetailViewController

+ Show EventDetailViewController, configured according to event data
+ Change `AppTransportSecurity` settings to allow arbitrary loads for
event and sponsored images

connienguyen committed:
Event Registration UI, Updated LoginManager

+ Update `LoginManager` to use `SocialNetworkAuthenticationStrategy`
for login
+ Update `LoginManager` unit test to match changes to using
`SocialNetworkAuthenticationStrategy`
+ Storyboard UI for `EventRegistrationViewController` + localized
keys/strings
+ Update view controllers using `LoginManager` to use new protocol
strategy

connienguyen committed:
Event Registration UI, Updated LoginManager

+ Update `LoginManager` to use `SocialNetworkAuthenticationStrategy`
for login
+ Update `LoginManager` unit test to match changes to using
`SocialNetworkAuthenticationStrategy`
+ Storyboard UI for `EventRegistrationViewController` + localized
keys/strings
+ Update view controllers using `LoginManager` to use new protocol
strategy

connienguyen committed:
UI for EventTableViewController, additional ETouchesAPIService funcions

+ Create `EventCell` to display event details in
`EventTableViewController`
+ Remove dead table cells
+ Extend `ETouchesAPIService to use PromiseKit and better mock real
eTouches API with URL parameters
+ Set up `ETouchesAPIService` to renew `accessToken` from eTouches ->
incomplete

connienguyen committed:
UI for EventTableViewController, additional ETouchesAPIService funcions

+ Create `EventCell` to display event details in
`EventTableViewController`
+ Remove dead table cells
+ Extend `ETouchesAPIService to use PromiseKit and better mock real
eTouches API with URL parameters
+ Set up `ETouchesAPIService` to renew `accessToken` from eTouches ->
incomplete

connienguyen committed:
Git rebase from `login-manual`

+ Include latest changes from `login-manual` including class comments
and updated unit tests

connienguyen committed:
Fix conflicts in `.pbxproj` file from git rebase

Could not push to remote repo without git error calling for a pull and
push

connienguyen committed:
Remove dead storyboard elements, refactor code

+ Removed unused Calendar VC from storyboard (using
EventTableViewController instead)
+ Moved cell configuration to EventCell
+ Use let `self` instead of let controller = self where possible

connienguyen committed:
Refactor ETouchesService and use EventTableViewController for Home

+ Separate ETouchesService file into smaller files including:
Constants+eTouches, ETouchesAccessTOken
+ Have HomeNavigationController use EventTableViewController instead of
placeholder vc for dev purposes
+ Added missing comments on classes
+ Removed *.orig from .gitignore

bphenriques commented:
@connienguyen Organize the filesystem please

https://github.com/venmo/synx

bphenriques commented:
You have a `VOLATests` that is not used. If so, delete it

bphenriques commented:
I would move the `Localizable.strings` to the `Resources` folder. I am always unsure if it's the best place but I believe so because they contain the app's strings for different languages

bphenriques commented:
I might only noticed now, but why we have a `ErrorStrings`? It would made more sense to me if it was of type Error or LocalizableError. 

If required. Add a convenience method `Logger.error(error: Error)`

bphenriques commented:
On the services layer you are repeating the creating of the promise several times, can you refactor the code so that only the particularities are provided to the creation of thr promises ? (url, parameters, type) ? :)

tip: protocol oriented programming + `associatedtype`

connienguyen committed:
Added missing comments, further separate ETouchesAPIService, Unittests

+ Updated `ErrorStrings` to `VLError`
+ Added missing comments to classes such as `Event`, `Location`,
`LoginManager`
+ Separated `ETouchesAPIService` further so that events related calls
are on one extension
+ Added helper enum `JSONFileNames` for testing as well as events
specific testing constants
+ Added default JSON responses to use for offline unittests
(`GetEventDetail.json`, `GetAvailableEvents.json`)
+ Set up unit tests for `ETouchesAPIService`
+ Added `Mockingjay` for stubbing `Alamofire` responses (not working
for some reason, may need help)

connienguyen committed:
Used `synx` to reorganize project files, added additional constants (`TextSize`)

connienguyen committed:
Passing unit tests for ETouchesAPIServiceUnitTests

+ Fixed unit tests to correctly use stubbed response by adjusting for
URL query parameters used for eTouches API

connienguyen committed:
PR changes: fixed Promise syntax, added constants, renamed variables

+ Simplified Promise syntax to not use `firstly` and to fix using
``self`` for viewcontroller.self
+ Added constants for “Y”/“N” for `VLCheckbox`
+ Renamed variables to be more descriptive
+ Use only one `baseURL`

connienguyen committed:
Added constants, modified Logger.error

+ Added constants throughout instead of magic values (both in VOLA and
VOLATests)
+ Moved localizable UI strings to constants
+ Modified `Logger` to take `Error` as an input instead of having to
use `.localizedDescription`
+ Adjusted affected Logger statements

connienguyen committed:
Created DefaultsManager, added unit tests for faulty server responses

+ Created `DefaultsManager` to handle setting and setting standard user
defaults values
+ Added unit tests to test `ETouchesAPIService` in cases with invalid
JSON responses or JSON responses with missing fields

connienguyen committed:
Week 3 Summary

Add table view controller for viewing events

Add view controller for viewing event details

Add view controller for event registration

Refactor LoginManager to use SocialNetworkingAuthenticationStrategy protocol

Created ETouchesAPIService to handle calls to eTouches API

Passing unit tests for ETouchesAPISerivce using mock responses from JSON files

Moved literal values to constants

Reorganized file structure using synx

Add DefaultsManager to have single point to retrieve data from user defaults
